packages/runtime/src/components/page/HeadSnippet.tsx

```diff
@@ -0,0 +1,122 @@
+import { Children, createElement, useEffect, useRef } from 'react'
+import { Snippet } from '../hooks/usePageSnippets'
+import { useServerInsertedHTML } from 'next/navigation'
+import parse from 'html-react-parser'
+import Head from 'next/head'
+import { useIsPagesRouter } from '../../next/hooks/use-is-pages-router'
+
+type Props = {
+  snippet: Snippet
+}
+
+export function HeadSnippet({ snippet }: Props) {
+  const isPagesRouter = useIsPagesRouter()
+
+  useEffect(() => {
+    if (isPagesRouter) {
+      return () => {
+        const snippetElements = getSnippetElementsFromDOM(snippet)
+        if (snippetElements.length > 0) return
+        cleanUpSnippet(snippet)
+      }
+    }
+
+    const snippetElements = getSnippetElementsFromDOM(snippet)
+    const isAlreadyInTheDOM = snippetElements.length > 0
+
+    if (isAlreadyInTheDOM) return
```

Comment:
Offline discussion: choosing the no-cleanup policy for server inserted scripts because this is basically what is happening with `next/head` already. Cleanups are still performed for snippets that are DOM managed - this brings the snippet execution between Pages Router / App Router 99% in sync.